src/Features/Core/Portable/ExtractClass/ExtractClassOptions.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.PullMemberUp;
+
+namespace Microsoft.CodeAnalysis.ExtractClass
+{
+    internal class ExtractClassOptions
+    {
+        public ExtractClassOptions(
+            string fileName,
+            string typeName,
+            bool sameFile,
+            ImmutableArray<ExtractClassMemberAnalysisResult> memberAnalysisResults)
+        {
+            FileName = fileName;
+            TypeName = typeName;
+            MemberAnalysisResults = memberAnalysisResults;
+            SameFile = sameFile;
+        }
+
+        public string FileName { get; }
+        public string TypeName { get; }
+        public ImmutableArray<ExtractClassMemberAnalysisResult> MemberAnalysisResults { get; }
+        public bool SameFile { get; }
+    }
+
+    internal class ExtractClassMemberAnalysisResult
+    {
+        /// <summary>
+        /// The member needs to be pulled up.
+        /// </summary>
+        public readonly ISymbol Member;
+
+        /// <summary>
+        /// Whether to make the member abstract when added to the new class
+        /// </summary>
+        public readonly bool MakeAbstract;
```

Comment:
can you use fields/props consistently in your data types?  i don't care which, just pick one :)